### PR TITLE
Change ContainerSE to derive from StructureSE

### PR DESCRIPTION
**Pull request by idolord**
switched CargoContainer and ContainerSE inheritance to structureItem and StructureSE respectively.
should slove the cargo weird behaviours (not loading) ...

### CODE REVIEW

**Comment by Almamu** [comment]
Can one of the admins verify this patch?

**Comment by jdhirst** [comment]
@Almamu you posted comments in this PR? I don't see them

**Comment by jdhirst** [comment]
!oktotest

**Comment by Almamu** [comment]
I started a review, you might need to expand it to see my comments.

**Comment by Almamu** [comment]
Sorry about the review thing, forgot to click the button to actually publish it -.-'

File: src/eve-server/system/Container.h [hunk 1]
```diff
@@ -163,8 +165,8 @@ class ContainerSE
     void Activate(int32 effectID);
     void Deactivate(int32 effectID);
     void AnchorContainer();
-    bool IsEmpty()                                      { return m_contRef->IsEmpty(); }
-    bool IsAnchored()                                   { return m_contRef->IsAnchored(); }
+    bool IsEmpty()                                      { return m_contRef->GetMyInventory()->IsEmpty(); }
```
[the remark this answers is not in the document]
Comment (jdhirst) [reply src/eve-server/system/Container.h:168]: @Almamu yeah, you're right, it should be derived from the base class Structure

File: src/eve-server/system/Container.cpp [hunk 2]
```diff
@@ -176,15 +176,29 @@ void CargoContainer::RemoveItem(InventoryItemRef iRef)
...
-            sLog.Warning( "CargoContainer::RemoveItem()", "Non-Cargo Container %u (type: %u) is empty and being deleted.", m_itemID, typeID() );
+            Delete();
+            break;
+        case EVEDB::invTypes::SmallSecureContainer:
+        case EVEDB::invTypes::SmallStandardContainer:
+        case EVEDB::invTypes::MediumSecureContainer:
+        case EVEDB::invTypes::MediumStandardContainer:
+        case EVEDB::invTypes::LargeStandardContainer:
+        case EVEDB::invTypes::LargeSecureContainer:
+            sLog.Warning("CargoContainer::RemoveItem()", "Launch Container %u is empty and being deleted.", m_itemID);
```
[the remark this answers is not in the document]
Comment (jdhirst) [reply src/eve-server/system/Container.cpp:197]: @Almamu indeed, it should only be the jetcans that are deleted like this

File: src/eve-server/system/Container.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (idolord) [reply src/eve-server/system/Container.h:168]: yeah reverted back in next commit

File: src/eve-server/system/Container.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (idolord) [reply src/eve-server/system/Container.cpp:197]: changed to indicate a 2 days timer if not anchored even tho anchoring isn't finished yet next commit maybe

**Comment by jdhirst** [comment]
Looks good to merge, great work @idolord !

**Pull request merged**